components/brave_talk/browser/brave_talk_frame_host.cc

```diff
@@ -0,0 +1,109 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_talk/browser/brave_talk_frame_host.h"
+
+#include <memory>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#include "absl/types/optional.h"
+#include "base/bind.h"
+#include "base/callback_forward.h"
+#include "base/logging.h"
+#include "brave/browser/brave_talk/brave_talk_service.h"
+#include "brave/browser/brave_talk/brave_talk_service_factory.h"
+#include "brave/components/brave_talk/common/features.h"
+#include "build/build_config.h"
+#include "chrome/browser/bad_message.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "content/browser/bad_message.h"
+#include "content/browser/renderer_host/frame_tree_node.h"
+#include "content/browser/renderer_host/navigation_controller_impl.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/web_contents.h"
+#include "third_party/blink/public/common/tokens/tokens.h"
+
+namespace brave_talk {
+class FrameFinder {
```

Comment:
This `FrameFinder` stuff can all be avoided if I use `RenderFrameHostImpl::FindAndVerifyChild`, which takes a `blink::FrameToken` directly. However, I vaguely remember that depending on stuff in `content/browser` is a bit of a nono. 

I'd be interested to hear someone who knows more's thoughts (like, maybe this is a bit of a smell).